### PR TITLE
add 281 institutions based on muamalat pdf

### PR DESCRIPTION
**Pull request by topekuno**
Salam geng saya mari dengan tolong besar daripada tolong sikit je last commit

Extract sebanyak mungkin dalam [pdf](https://www.muamalat.com.my/wp-content/uploads/2023/10/LIST-OF-QR-CODE-FOR-E-DERMA.pdf)

Extracted data mainly from QR image itself but I think it's good enough. Can get all sorts of data from duitnow QR and extrapolate from there.

Check for uniqueness too based on `qrContent` from current institutions.ts at this point of time so no duplicated data is added

Many trials and error but happy with the outcome.

You can check out the scripts at my repo if you wanna have a look and use them kalau nak just bagi credit je atau belanja teh o ais satu.
https://github.com/topekuno/sedekah-qr-extractor

Okay tido lu pape roger

### CODE REVIEW

File: app/data/institutions.ts [hunk 1]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "surau",
+    state: "W.P. Kuala Lumpur",
+    city: "Kuala Lumpur",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000001031001230919085204866153034585802MY5916SURAU AS SAKINAH6012KUALA LUMPUR6105501006304B99A",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 280,
+    name: "Masjid Damansara Perdana",
```
Comment (khrnchn) [inline app/data/institutions.ts:3362]: image

File: app/data/institutions.ts [hunk 2]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "surau",
+    state: "W.P. Kuala Lumpur",
+    city: "Taman Minang Kl",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000318031001692066535204866153034585802MY5916SURAU NURUL IMAN6015TAMAN MINANG KL61055600063048622",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 283,
+    name: "Masjid Nurrusaadah ",
```
Comment (khrnchn) [inline app/data/institutions.ts:3392]:
```suggestion
    name: "Masjid Nurrusaadah",
```

File: app/data/institutions.ts [hunk 3]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    city: "Cheras",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000001196031001238230665204866153034585802MY5918SURAU AL-MUKHLISIN6006CHERAS6105560006304797E",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 286,
+    name: "Surau At-Taqwa",
+    category: "surau",
+    state: "Kedah",
```
Comment (khrnchn) [inline app/data/institutions.ts:3424]: state kedah but city cheras

File: app/data/institutions.ts [hunk 4]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "surau",
+    state: "W.P. Kuala Lumpur",
+    city: "Segambut",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000194031001226195235204866153034585802MY5919SURAU AL-MUQARRABIN6008SEGAMBUT610551200630473AD",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 289,
+    name: "Per Seruan Islam Sel W.p",
```
Comment (khrnchn) [inline app/data/institutions.ts:3452]:
```suggestion
    name: "Persekutuan Seruan Islam Selangor & Wilayah Persekutuan (Jam'iyah)",
```

File: app/data/institutions.ts [hunk 5]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "mosque",
+    state: "W.P. Kuala Lumpur",
+    city: "Kuala Lumpur",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000103031001925422105204866153034585802MY5923MASJID ZAHID BIN THABIT6012KUALA LUMPUR6105560006304C4A8",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 294,
+    name: "Masjid Ibn Khaldun",
```
Comment (khrnchn) [inline app/data/institutions.ts:3502]: already exists on website

File: app/data/institutions.ts [hunk 6]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "mosque",
+    state: "Selangor",
+    city: "Hulu Kelang",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000224031001938192125204866153034585802MY5919MASJID UKAY PERDANA6011HULU KELANG61056800063045E90",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 300,
+    name: "Masjid A.ubaidah Al Jarrah",
```
Comment (khrnchn) [inline app/data/institutions.ts:3562]:
```suggestion
    name: "Masjid Abu Ubaidah Al Jarrah",
```

File: app/data/institutions.ts [hunk 7]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "others",
+    state: "W.P. Kuala Lumpur",
+    city: "Setapak",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000904031001333224145204533153034585802MY5924PER PNDDK TMN DANAU KOTA6007SETAPAK610553300630449F7",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 306,
+    name: "Masjid Al-Umm",
```
Comment (khrnchn) [inline app/data/institutions.ts:3622]: already exists on website

File: app/data/institutions.ts [hunk 8]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "surau",
+    state: "Selangor",
+    city: "Batu Caves",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000250031001939216615204866153034585802MY5918SURAU AL-IKHWAN TS6010BATU CAVES6105681006304EB4C",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 310,
+    name: "Surau Al Hidayah T.sunway",
```
Comment (khrnchn) [inline app/data/institutions.ts:3662]:
```suggestion
    name: "Surau Al Hidayah Taman Sunway",
```

**Comment by khrnchn** [comment]
@topekuno fuh, thanks for the huge effort! thank you for sharing the script too. that's a LOT of data @@

i have reviewed some of the newly added data; some cleanup is needed. will continue from time to time. 

@Kyziq kalau rajin tengok ni

File: app/data/institutions.ts [hunk 9]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "surau",
+    state: "Johor",
+    city: "Johor",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000716031001965652055204866153034585802MY5911SURAU EHSAN6005JOHOR61058130063040ED1",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 591,
+    name: "M T Al Quran As Sunnah",
```
Comment (khrnchn) [inline app/data/institutions.ts:6472]:
```suggestion
    name: "Maahad Tahfiz Al Quran Wa As Sunnah",
```

File: app/data/institutions.ts [hunk 10]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "surau",
+    state: "Johor",
+    city: "Muar",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000665031001975056005204866153034585802MY5925SURAU AL BARAKAH BAKRI B26004MUAR6105840006304F5A4",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 589,
+    name: "Mtaq At Tanwiriah",
```
Comment (khrnchn) [inline app/data/institutions.ts:6452]:
```suggestion
    name: "Madrasah Tahfiz Al Quran At Tanwiriah",
```

File: app/data/institutions.ts [hunk 11]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "others",
+    state: "Johor",
+    city: "Segamat",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000343031001374343995204866153034585802MY5924B.TDBIR MJ JAMEK B.SIPUT6007SEGAMAT61058500063048893",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 583,
+    name: "Pembangunan Sur. An-Nur D.skudai",
```
Comment (khrnchn) [inline app/data/institutions.ts:6392]:
```suggestion
    name: "Pembangunan Surau An-Nur Desa Skudai",
```

File: app/data/institutions.ts (hunk 3 above)
Comment (topekuno) [reply app/data/institutions.ts:3424]: seems to be an edge case here when processing the postcode via the script should be state W.P. Kuala Lumpur checked in pdf

File: app/data/institutions.ts [hunk 12]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "others",
+    state: "Johor",
+    city: "Segamat",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000340031001978968605204300053034585802MY5922MJD JMK TENANG STATION6007SEGAMAT61058503063048C34",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 581,
+    name: "Masjid Jmk Kampung Paya Pulai",
```
Comment (khrnchn) [inline app/data/institutions.ts:6372]:
```suggestion
    name: "Masjid Jamek Kampung Paya Pulai",
```

File: app/data/institutions.ts [hunk 13]
```diff
@@ -3347,4 +3347,3544 @@ export const institutions: Institution[] = [
...
+    category: "surau",
+    state: "Johor",
+    city: "Segamat",
+    qrImage: "",
+    qrContent: "00020201021126610014A000000615000101065641670215QRMID0000000339031001376523505204866153034585802MY5920SURAU HASAN AL BASRI6007SEGAMAT61058500063048491",
+    supportedPayment: ["duitnow", "tng"],
+  },
+  {
+    id: 580,
+    name: "Masjid Jmk Tenang Station",
```
Comment (khrnchn) [inline app/data/institutions.ts:6362]:
```suggestion
    name: "Masjid Jamek Tenang Station",
```

**Comment by Kyziq** [comment]
Insane work and effort nii 🤩Jadinya the whole .pdf is done, just need to fine-tune macam cari duplicate, better naming for the name and city (easier to identify)

eg:
city Masjid Al-Mukmin Kampung Bukit Durian to Ayer Molek

**Comment by Kyziq** [comment]
@khrnchn macam mana nak contribute dekat PR ni? Kalau commit nanti appear dekat sini, kena ada access eh

**Comment by khrnchn** [comment]
> @khrnchn macam mana nak contribute dekat PR ni? Kalau commit nanti appear dekat sini, kena ada access eh

checkout this branch then work in local kot?

**Comment by nrmnqdds** [comment]
i think nak merge dulu the cleanup kemudian pun okay je since aku dah buat remove duplicates on frontend side
image

**Pull request merged**